SalesforceFoundation/EDA
# [EDA] Prevent deletion of a Term Record when there is a Related Child Term Record

# Critical Changes

# Changes
## Update to Ensure Data Integrity
To ensure greater data integrity, we've added functionality to prevent the deletion of Term records that have related Term records. This change is for internal-use only and is disabled at this time. You won't see any difference in functionality.
**Note:** We strongly recommend that EDA Admins don't manage Hierarchy Custom Settings directly and, instead, use the EDA Settings UI.

# Issues Closed
Closes #1124 

# New Metadata

# Deleted Metadata

# Testing Notes
Scenario 1: “Prevent Term Deletion” checkbox is enabled and Term record have related Term Grade, Course Connection, or Course Offering Schedule or Term records.
1) Check “Prevent Term Deletion” checkbox in Hierarchy Settings.
2) Create 2 Term records
3) Associate 1 Term record to another.
4) Delete the parent Term record.
5) You should receive a pop up error that you cannot delete this Term. See https://salesforce.quip.com/fXSbAE7hHWV8#LPKACAsjDGR for the error message.

Scenario 2: “Prevent Term Deletion” checkbox is disabled and Term record have related Term Grade, Course Connection, or Course Offering Schedule or Term records.
1) UnCheck “Prevent Term Deletion” checkbox in Hierarchy Settings.
2) Create 2 Term records
3) Associate 1 Term record to another.
4) Delete the parent Term record.
5) The Parent Term should be deleted

Scenario 3: “Prevent Term Deletion” checkbox is disabled and Term record have related Term Grade, Course Connection, or Course Offering Schedule or Term records.
1) Uncheck “Prevent Term Deletion” checkbox in Hierarchy Settings.
2) Create 2 Term record
3) Associate 1 Term record to another.
4) Delete the Term record
5) You should be able to delete the Parent Term record

Scenario 4: “Prevent Term Deletion” checkbox is disabled and Term record does not have related Term Grade, Course Connection, or Course Offering Schedule or Term records.
UnCheck “Prevent Term Deletion” checkbox in Hierarchy Settings.
1) Create a Term record
2) Delete the Term record
3) You should be able to delete the Parent Term record
